apps/io-wallet-user-func/src/infra/email.ts

```diff
@@ -0,0 +1,92 @@
+import { AuthProfileApiConfig, MailConfig } from "@/app/config";
+import { EmailNotificationService, SendEmailNotificationParams } from "@/email";
+import {
+  getMailerTransporter,
+  sendMail,
+} from "@pagopa/io-functions-commons/dist/src/mailer";
+import { EmailString, FiscalCode } from "@pagopa/ts-commons/lib/strings";
+import { flow, pipe } from "fp-ts/function";
+import * as TE from "fp-ts/lib/TaskEither";
+import * as t from "io-ts";
+
+export class EmailNotificationServiceClient
+  implements EmailNotificationService
+{
+  #authProfileApiConfig: AuthProfileApiConfig;
+  #mailConfig: MailConfig;
+
+  getUserEmail = (fiscalCode: FiscalCode) =>
+    pipe(
+      TE.tryCatch(
+        async () => {
+          const result = await fetch(
+            new URL(
+              `/api/v1/profiles/${fiscalCode}`,
+              this.#authProfileApiConfig.baseURL,
+            ),
+            {
+              headers: {
+                "x-functions-key": this.#authProfileApiConfig.apiKey,
+              },
+              method: "GET",
+              signal: AbortSignal.timeout(
+                this.#authProfileApiConfig.httpRequestTimeout,
+              ),
+            },
+          );
+          if (!result.ok) {
+            throw new Error(JSON.stringify(await result.json()));
```

Comment:
@silvicir WDYT? It doesn't necessarily have to be JSON. By the way it is converted to string anyway
```suggestion
            throw new Error(await result.text());
```

Reply:
it's fine with me as long as you're sure it works this way